### PR TITLE
Make publish script fail when `file` is missing

### PR DESCRIPTION
**Pull request by jez**
### Motivation

This will make publish.sh fail fast when the `file` command isn't
available. As far as I can tell, there's no way to make the bash
pattern:

    some_good_cmd "$(some_bad_command)"

fail fast by enabling a flag. In every case it will need to be
refactored to

    x="$(some_bad_command)"
    some_good_cmd "$x"

which as a Bash enthusiast is deeply saddening. shellcheck doesn't warn
for this, no future version of bash solves this, etc. I don't think it's
worth patently avoiding this, because it can make things more verbose
and coming up with names can be hard, but it's something to be aware of.

### Test plan

This will fail if we merge to master, because this change is hidden on branch
builds behind a `$dryrun` check.

The only machine affected right now is the non-elastic Linux build machine.
We'll need to make sure that `file` is installed there before proceeding.

### CODE REVIEW

File: .buildkite/tools/gh-release.sh [hunk 1]
```diff
@@ -82,9 +82,10 @@ response=$(
 
 upload_url="$(echo "$response" | jq -r .upload_url | sed -e "s/{?name,label}//")"
 
+content_type="Content-Type:$(file --mime-type -b "${asset}")"
```
Comment (DarkDimius) [inline .buildkite/tools/gh-release.sh]: can we make this into a nicer error?
Like 
https://github.com/sorbet/sorbet/blob/5a9d35b314d4680797f8a7a79d5ec22427240390/tools/toolchain/webasm-linux/emcc.sh#L11

**Comment by DarkDimius** [comment]
ooc: why would `file` be missing?

**Comment by jez** [comment]
> ooc: why would `file` be missing?

I have no idea. I'll have to look into it tomorrow. But like, for a long time we **were** successfully releasing to GitHub Releases. I'm not sure what changed.

File: .buildkite/tools/gh-release.sh [hunk 2]
```diff
@@ -43,6 +43,10 @@ fi
 
 [ "$2" != "" ] || (usage; exit 1);
 
+if ! command file &> /dev/null; then
+  echo "This script requires the 'file' command, but it was not in PATH"
```
Comment (DarkDimius) [inline .buildkite/tools/gh-release.sh]: should we make it fail?

Comment (jez) [reply .buildkite/tools/gh-release.sh]: ack ty for catching this.

**Pull request merged**